src/lib/rules/player.hh

```diff
@@ -74,12 +78,22 @@ struct Players final : utils::IBufferizable
         return ss.str();
     }
 
-    size_t size() const { return players.size(); }
+    void add(std::shared_ptr<Player> player) { players_.push_back(player); }
 
-    std::vector<Player_sptr> players;
-};
+    // Sort players by identifiers
+    void sort()
+    {
+        std::sort(players_.begin(), players_.end(),
+                  [](const auto& a, const auto& b) { return a->id < b->id; });
+    }
+
+    size_t size() const { return players_.size(); }
 
-using Players_sptr = std::shared_ptr<Players>;
+    const PlayersVector& all() const { return players_; }
```

Comment:
Optional: implement the iterator "trait" instead? This looks like this method is here to hide the fact this class is just a proxy for `vector<Player>`.